slackclient/client.py

```diff
@@ -132,13 +132,16 @@ def rtm_read(self):
         # in the future, this should handle some events internally i.e. channel
         # creation
         if self.server:
-            json_data = self.server.websocket_safe_read()
             data = []
-            if json_data != '':
-                for d in json_data.split('\n'):
-                    data.append(json.loads(d))
-            for item in data:
-                self.process_changes(item)
+            while True:
```

Comment:
why do you need that `while True` if it always exits on first iteration?  I'd think the following would be equivalent:

```
def rtm_read(self):
    if not server:
        raise SlackNotConnected
    json_data = self.server.websocket_safe_read()
    data = [json.loads(d) for d in json_data.splitlines()]
    for item in data:
        self.process_changes(item)
    return data
```

Reply:
The while True loop is to call websocket_safe_read multiple times because at the moment websocket_safe_read method only returns the first message, not all available messages. 

Looking at it, perhaps the change should be to websocket_safe_read instead, as it looks like the intention is to return all available messages from there?

Reply:
> Looking at it, perhaps the change should be to websocket_safe_read
>  instead, as it looks like the intention is to return all 
>  available messages from there?

I think that's right... `websocket_safe_read()` needs fixing

Reply:
Related: #231